### PR TITLE
Update Scala versions

### PR DESCRIPTION
**Pull request by sideeffffect**
Continuation of https://github.com/portable-scala/portable-scala-reflect/pull/24
/cc @LeeTibbert @sjrd

### CODE REVIEW

File: project/plugins.sbt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 val scalaJSVersion =
-  Option(System.getenv("SCALAJS_VERSION")).filter(_ != "").getOrElse("1.0.0")
+  Option(System.getenv("SCALAJS_VERSION")).filter(_ != "").getOrElse("1.0.1")
 
 addSbtPlugin("org.scala-js" % "sbt-scalajs" % scalaJSVersion)
-addSbtPlugin("org.portable-scala" % "sbt-scalajs-crossproject" % "0.6.0")
+addSbtPlugin("org.portable-scala" % "sbt-scalajs-crossproject" % "1.0.0")
 
-addSbtPlugin("com.typesafe" % "sbt-mima-plugin" % "0.3.0")
+addSbtPlugin("com.typesafe" % "sbt-mima-plugin" % "0.8.1")
```
Comment (sjrd) [inline project/plugins.sbt:7]: It seems the MiMa upgrade kills Scala 2.10.

Comment (sideeffffect) [reply project/plugins.sbt:7]: For the record, do we want to keep Scala 2.10? Even old Spark is on 2.11 and new is on 2.12.
The last MiMa to support 2.10 is 0.3.0
https://github.com/lightbend/mima/releases/tag/0.3.0
https://github.com/lightbend/mima/releases/tag/0.4.0

Not using the new MiMa prevents us (for some reason :shrug: ) from using the new sbt.

Comment (sjrd) [reply project/plugins.sbt:7]: Is there any reason to use the new sbt?

**Comment by LeeTibbert** [comment]
@sideeffffect Travis is all green.  Looked like you have cracked the nut!  Thank you.

I will base my sandbox work on this, especially `.travis.xml` and hope to be
ready with the PR adding SN support  soon after this is merged.

**Comment by sjrd** [comment]
OK, let's drop 2.10, but we must do it properly. It should be in its own PR, as it is a feature drop, with two commits:
1. First commit: Bump the version number at https://github.com/portable-scala/portable-scala-reflect/blob/304fe9ca08e464afb29849ebae54b1392e452ccc/build.sbt#L9 to 1.1.0-SNAPSHOT
2. Second commit: Drop support for 2.10. This should also remove the `TestPlatform.scala` in https://github.com/portable-scala/portable-scala-reflect/tree/master/js/src/test/scala/org/portablescala/reflect and https://github.com/portable-scala/portable-scala-reflect/tree/master/jvm/src/test/scala/org/portablescala/reflect

**Review by sjrd — CHANGES_REQUESTED** [review]
See above.

**Comment by sideeffffect** [comment]
Dropping 2.10 in https://github.com/portable-scala/portable-scala-reflect/pull/29

**Pull request closed without merging**